### PR TITLE
Retry buffer size cleanup

### PR DESCRIPTION
**Pull request by pepone**
This small PR cleanup the checks and logic for RetryBufferSizeMax and RetryRequestSizeMax

### CODE REVIEW

File: csharp/src/Ice/Proxy.cs
Comment (bentoi) [inline csharp/src/Ice/Proxy.cs:460]: If `releaseRequestAfterSent` is true but the request isn't sent... can't we still retry? Won't this check prevent such a retry?

**Review by bentoi — APPROVED** [review]
Added a comment. I'll wait for this PR to be merged before to merge by branch. I'll revisit proxy InvokeAsync code, it should be simpler now that InvokeAsync makes different calls for sending the request and receiving the response.

File: csharp/src/Ice/Proxy.cs
Comment (pepone) [reply csharp/src/Ice/Proxy.cs:460]: good catch, we should check for sent here to allow the retry

Comment (pepone) [reply csharp/src/Ice/Proxy.cs]: fixed

**Review by bernardnormier — APPROVED** [review]

**Pull request merged**
